CMakeLists.txt

```diff
@@ -88,6 +88,8 @@ if(NOT HARDINFO_NOSYNC)
 	pkg_check_modules(LIBSOUP libsoup-2.4>=2.24)
 endif()
 
+pkg_check_modules(X11 REQUIRED x11)
```

Comment:
Is requiring X11 the way to go? GTK3 can use Wayland, and it's not really a good idea to require X11 in that case.

Reply:
but older machines need X11 and does not work with Wailand, some distributions does will not include never Wayland and seems for now, if wayland will be used , x11 will be used as base, such as debian 9

Reply:
It is possible to have a machine with no xserver, nor xrandr, nor xdpyinfo, that still has an X-client application running on it. I was running on a x-less debian machine with `ssh -X ...` and it left the whole display information empty, even though X is clearly being used. The xlib dep is for a fallback of getting information directly from X. Otherwise, (for example, in Debian) we need to make sure that x11-xserver-utils is an install dependency of the hardinfo package, so that we know that xrandr and xdpyinfo will be available.

Reply:
Also, we are guaranteed to have Xlib available if we have GDK. Honestly, we might try and get all the X information directly via Xlib instead of relying on `xrandr` and/or `xdpyinfo`.

Reply:
umm so then now X11 its a new depend, so about the reading of scsi older kernels vs newer kernels.. ? may runnign good hardware only work with older kernel..  to the list of TODO (theres at leas one  in the issues) we must added the new depens and runtime-depends and i'll prepare a README introduction update and compilation info how to

Reply:
My reply didn't get added to this thread, but instead the PR thread, so I'll duplicate it here.
X11 was always required via GDK, only now the Xlib headers are needed to build. This really doesn't change anything for the binary, it just lets the cmake magic work to find the right linking config.